scripts/koji/koji_build.py

```diff
@@ -59,12 +60,23 @@ def local_branch(branch):
     finally:
         subprocess.check_call(['git', 'checkout', prev_branch])
         subprocess.check_call(['git', 'branch', '-D', branch])
-        subprocess.check_call(['git', 'push', '--delete', 'origin', branch])
+
+def is_old_branch(b):
+    branch_time = datetime.strptime(b.split('/')[-1], TIME_FORMAT)
+    return branch_time < datetime.now() - timedelta(hours=3)
 
 def push_bumped_release(git_repo, trigram):
-    uuid = uuid4()
-    branch = f'koji/test/{uuid}'
+    t = datetime.now().strftime(TIME_FORMAT)
+    branch = f'koji/test/{trigram}/{t}'
     with cd(git_repo), local_branch(branch):
+        # clean up the old branches
+        subprocess.check_call(['git', 'fetch'])
+        remote_branches = subprocess.check_output(['git', 'branch', '-rl', 'origin/koji/test/*/*']).decode('utf8').splitlines()
+        remote_branches = [b.strip()[len('origin/'):] for b in remote_branches]
+        old_branches = [b for b in remote_branches if is_old_branch(b)]
+        for b in old_branches:
+            print(f"removing outdated remote branch {b}", flush=True)
+            subprocess.check_call(['git', 'push', '--delete', 'origin', b])
```

Comment:
I'm dubious on the feature:
- either we allow to run several such test builds concurrently (which I think is a good idea, or at least I don't really see a reason not to), and then we don't want to destroy any branches
- or we don't allow it, and we might as well just use a unique branch name for all such tests

Further more, there should be very little space for such branches to linger, with context managers doing their job - only killing the script in a way python cannot recover should leave any such trace behind.

Reply:
You are encouraged to use a branch, we just won't pollute it for the build.

Reply:
we have to keep the remote branch if we want to run with `--nowait`. The context manager won't do the job in that case, because it's gone long before the build is done (or even started)